addons/payment_ogone/models/payment_acquirer.py

```diff
@@ -85,14 +85,14 @@ def _ogone_generate_signature(self, values, incoming=True, format_keys=False):
         def _filter_key(_key):
             return not incoming or _key in VALID_KEYS
 
-        key = self.ogone_shakey_in if incoming else self.ogone_shakey_out
+        key = self.ogone_shakey_out if incoming else self.ogone_shakey_in  # Swapped for Ogone's POV
         if format_keys:
             formatted_items = [(k.upper().replace('_', '.'), v) for k, v in values.items()]
         else:
             formatted_items = [(k.upper(), v) for k, v in values.items()]
         sorted_items = sorted(formatted_items)
         signing_string = ''.join(f'{k}={v}{key}' for k, v in sorted_items if _filter_key(k) and v)
-        shasign = sha256(signing_string.encode("utf-8")).hexdigest()
+        shasign = sha1(signing_string.encode("utf-8")).hexdigest()
```

Comment:
while you're touching the line you can simplify (utf-8 is the default)

```suggestion
        shasign = sha1(signing_string.encode()).hexdigest()
```